processor/metrics.go

```diff
@@ -172,10 +172,30 @@ func ServiceNameForResource(resource data.Resource) string {
 	return service.StringVal()
 }
 
+// ServiceNameForTraceData gets the service name for a TraceData.
+func ServiceNameForTraceData(traceData data.TraceData) (serviceName string) {
+	resourceSpans := traceData.ResourceSpans()
+	for i := 0; i < resourceSpans.Len(); i++ {
+		sn := ServiceNameForResource(resourceSpans.At(i).Resource())
+		if serviceName == "" {
+			serviceName = sn
+		} else if serviceName != sn {
+
+			// If there are more then one resources in the TraceData batch "<multiple-services>"
+			// keyword is set. To avoid this, batch processor should be used.
+			serviceName = "<multiple-services>"
+			return
+		}
+	}
+	return
+}
```

Comment:
I think we should simply report the metric for every service name.

Reply:
but it's not going to be accurate. for example, if we have a batch with two services, 2 spans from `service1` and 3 spans from `service2`, we will report:
```
otelcol_batches_received{processor="queued_retry",service="service1"} 5
otelcol_batches_received{processor="queued_retry",service="service2"} 5
```

Reply:
no, we should report:
otelcol_batches_received{processor="queued_retry",service="service1"} 2
otelcol_batches_received{processor="queued_retry",service="service2"} 3

Reply:
right. but it's not possible for `sent` and `dropped` metrics, otherwise we wound need to break down the batch before sending to downstream

Reply:
I am confused why is not possible. Can you clarify?

Reply:
Trying to explain why it's not possible I realized that it's very possible :)

Before sending to downsteam we can save how many spans each service has, then based on the downstream response we emit dropped or sent metrics for all the services

I'll add this change.

Reply:
@bogdandrutu I can do what I described for span count metrics ^ But for batches it seems to be actually impossible because one batch (+1 to metric counter) now can contain more than one service. 

In my example I confused batch and span count metrics: if we get a batch with two services, 2 spans from service1 and 3 spans from service2 we can expose metrics:
```
otelcol_spans_received{processor="queued_retry",service="service1"} 2
otelcol_spans_received{processor="queued_retry",service="service2"} 3
```
but cannot expose `otelcol_batches_received` with service tag, so we can either  put `"<multiple-services>"` as the service tag value (in that case i'd do the same for span count metrics as well). Or we can just drop service tag and expose:
```
otelcol_batches_received{processor="queued_retry",source_format="oc_trace"} 
```

What do you think?

Reply:
@bogdandrutu I implemented the approach I proposed above, appeared to be a bit tricky. Please take a look once you have a chance